org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/BasicProjectRegistry.java

```diff
@@ -94,18 +94,16 @@ protected final void replaceWith(BasicProjectRegistry other) {
    */
   @SuppressWarnings({"unchecked", "rawtypes"})
   private static void copy(Map from, Map to) {
-    for(Map.Entry entry : (Set<Map.Entry>) from.entrySet()) {
-      Object value = entry.getValue();
+    from.forEach((key, value) -> {
```

Comment:
I don't no what this method does but the lamda do not make it more clear I think... should it be some kind of deepcopy of a map?

Reply:
> I don't no what this method does but the lamda do not make it more clear I think... should it be some kind of deepcopy of a map?

I have no details about that either, but yes I think that is a deep copy of Maps/Sets that does not copy the elements itself.
But personally I think the forEach makes it more clear, because no casting and key/value getting is necessary, if one knows/is used to that method.